CMSgov/easi-app
# [EASI-3361] TRB admin note frontend updates

# EASI-3361

[Figma](https://www.figma.com/file/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?type=design&node-id=5344-49549&mode=design&t=aFVNSK7eMyzvmahU-0)

## Changes and Description

- Created `trbAdminNoteUpdates` feature flag to hide frontend work until backend work is completed
- Added conditional fields to form in `AddNote` component for category specific data
  - Initial request form, supporting documents, and advice letter categories have additional fields
  - Mutation logic is in `useAddNote` custom hook (src/views/TechnicalAssistance/AdminHome/AddNote/useAddNote.ts)
- Rendered additional field values when notes are displayed
- Style updates to notes admin view and modal
- Updated unit tests

**Note:** related backend work is still in progress. Schema work was done in EASI-3360, and remaining backend work will be completed in EASI-3362.



## How to test this change

- Enable `trbAdminNoteUpdates` feature flag
- Go to /trb/:trbRequestId/request
  - Click Add a Note button to test form and notes view in modal
- Go to /trb/:trbRequestId/notes
  - Make sure notes display matches Figma 
  - Click "Add a note" button to test non-modal form


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
